Fix missing library when linking GLFW3

catageek opened the pull request:
This may concern only gcc users. It should not bother other compilers.

catageek committed:
Fix missing library when linking GLFW3

wrongu commented:
I had the same problem with linking X libraries, but with danix's help I was able to fix it by removing libglfw.a and recompiling it as a .so.
(cmake -DBUILD_SHARED_LIBS=ON)

catageek committed:
Move the set line to be excluded from the Win32 section

catageek committed:
Remove the static library preference instead of trying to link against
it.

To build Sigma, glfw3 must be built from source using the command:

cmake -DBUILD_SHARED_LIBS=ON
